app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSponsorApiController.php

```diff
@@ -190,7 +215,18 @@ protected function deleteChild(Summit $summit, $child_id): void
 
     protected function getChildFromSummit(Summit $summit, $child_id): ?IEntity
     {
-        return $summit->getSummitSponsorById($child_id);
+        $current_member = $this->resource_server_context->getCurrentUser();
+        $sponsor = $summit->getSummitSponsorById($child_id);
+
+        if (!is_null($current_member) &&
```

Comment:
this condition seems too complicate lets simplify it
like
if(is_null($sponsor)) return null;
also asking for  $current_member->belongsToGroup(IGroup::Sponsors) is redundant bc
that should be asked inside $current_member->hasSponsorMembershipsFor($sponsor)